src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Scalar.cs

```diff
@@ -895,6 +1259,24 @@ public static bool ObjectEquals(T left, T right)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static T Round(T value)
+        {
+            if (typeof(T) == typeof(double))
+            {
+                return (T)(object)Math.Round((double)(object)value);
+            }
+            else if (typeof(T) == typeof(float))
+            {
+                return (T)(object)MathF.Round((float)(object)value);
+            }
+            else
+            {
+                ThrowHelper.ThrowNotSupportedException(ExceptionResource.Arg_TypeNotSupported);
```

Comment:
How do we end up here? I thought I saw that the Round public entrypoints were double / float-specific rather than generic?

Reply:
`ISimdVector<TSelf, T>` has `Round`, `Ceiling`, and a few other APIs generically right now. They no-op for integer types.